control_software/scripts/hera_fem_switch_loop.py

```diff
@@ -0,0 +1,239 @@
+else:
+    # py3
+    from plotly import graph_objects as go, subplots, io as pio
+
+logger = helpers.add_default_log_handlers(logging.getLogger(__name__),
+                                          fglevel=logging.NOTSET)
+
+
+def main(redishost='redishost', hostname=None, antenna_input=None,
+         integration_time=None, do_not_initialize=False):
+    if hostname is None:
+        raise ValueError("Must specify a hostname or list of hostames. "
+                         "To run on all Snaps set hostame='all'")
+    # not actually using the connection
+    # but will bw used tempo manipulate the monitoring state
+    # and find all hosts if needed.
+    c = HeraCorrelator(redishost=redishost,
+                       passive=True)
+    if hostname:
+        if not isinstance(hostname, (list, tuple)):
+            hostname = [hostname]
+
+    hostname = sorted(hostname)
+
+    if hostname == ['all'] and not antenna_input:
+        logger.info("Using all SNAPs from redis configuration")
+        SNAP_HOST = sorted(list(c.config['fengines'].keys()))
+        ANTENNA_INPUT = np.arange(0, 6)
+    elif hostname != ['all'] and not antenna_input:
+        SNAP_HOST = hostname
+        ANTENNA_INPUT = np.arange(0, 6)
+    else:
+        SNAP_HOST = hostname
+        ANTENNA_INPUT = antenna_input
+
+    correlations = {}
+    for host in SNAP_HOST:
+        # Disable the monitoring process for 5 minutes,
+        # to stop it interfering with this script
+        c.disable_monitoring(300, wait=True)
+
+        snap = SnapFengine(host, redishost=redishost)
+
+        if not snap.is_programmed():
+            snap.fpga.transport.prog_user_image()
+        if not do_not_initialize:
+            snap.initialize()
+
+        if integration_time is not None:
+            acc_len = int(integration_time * 250e6
+                          // (snap.corr.nchans * snap.corr.spec_per_acc)
+                          )
+            if not acc_len == snap.corr.get_acc_len():
+                snap.corr.set_acc_len(acc_len)
+
+        host_group = correlations.setdefault(host, {})
+
+        for antenna_ind in ANTENNA_INPUT:
+            logger.info("Running FEM switch loop "
+                        "for {host} antenna {ant}".format(host=host,
+                                                          ant=antenna_ind)
+                        )
+
+            c.disable_monitoring(60, wait=True)
+            fem_ind = antenna_ind / 2
+            fem_pol = antenna_ind % 2
+            fem = snap.fems[fem_ind]
+            current_state = fem.switch()
+
+            ant_group = host_group.setdefault(antenna_ind, {})
+
+            for state in ['noise', 'load', 'antenna']:
+                fem.switch(name=state)
+                autocorr = snap.corr.get_new_corr(antenna_ind, antenna_ind).real
+
+                eq_coeff = snap.eq.get_coeffs(antenna_ind)
+                ant_group[state] = autocorr / eq_coeff**2
```

Comment:
Can you please add an option to not do the EQ divide out. It makes it very difficult to tell what's going on in digital land.

Reply:
Thanks!